service/models.py

```diff
@@ -87,6 +80,15 @@ def all(cls):
         cls.logger.info('Processing all suppliers')
         return cls.objects()
 
+    @classmethod
+    def delete(cla, supplier_id)
+        """ Delete a promotions by it's ID """
```

Comment:
That seems like it should be "supplier", not "promotion"

Reply:
Fixed, thanks!